TheMovieDatabase/ViewControllers/DetailedMovie/View/DetailedMovieViewController.swift

```diff
@@ -225,7 +225,7 @@ final class DetailedMovieViewController: UIViewController {
 
     @objc
     private func likeTapped() {
-        if Locksmith.getSessionId() != nil {
+        if authorizationService.getSessionId() != nil {
```

Comment:
If you don't call Locksmith doesn't mean that ViewController don't know about the session-id)

Could you try to move all the logic connected to adding/removing from favorites (both via internet and database) to the method like `setFavorite(to: Bool, movieId: Int, completion: (Void) -> ())`

I know it's strange to show `favoriteActivityIndicator` when adding to database, but you'll probably won't notice it cause it would be fast call of startAnimating and endAnimating